src/Orleans/Configuration/ClientConfiguration.cs

```diff
@@ -74,6 +74,13 @@ public enum GatewayProviderType
         public string DataConnectionString { get; set; }
 
         /// <summary>
+        /// the id of the cluster to which this client is connecting. 
+        /// This argument is optional, but is needed in order for the client to receive
+        /// observer messages from different clusters.
+        /// </summary>
+        public string ClusterId { get; set; }
```

Comment:
the client is already talking to a particular cluster, why does the client need to configure it instead of being implied on the silo side, when there is a multi cluster network?